Automattic/wp-calypso
# Plans 2023: Update WooCommerce plans to use PlansFeaturesMain and intent system

Fixes https://github.com/Automattic/wp-calypso/issues/78472
Parent: https://github.com/Automattic/wp-calypso/pull/78731

> WooExpress currently imports and uses plans-features-2023-grid directly (presumably to render a tailored plans mix). This should be updated to go through PlansFeaturesMain in line with everything else, especially since the introduction of "intent" system in https://github.com/Automattic/wp-calypso/pull/77667.
> 
> plans-features-2023-grid should/will go through a series of "simplifications" next to become a proper package, and this may be the main blocker from doing so.

## Proposed Changes

- WooExpress plans page is now synced along with the rest to use `PlansFeaturesMain` and the new "intent" system
- ~`hideUnavailableFeatures` used previously also removed as it's redundant now in new framework~

### Need to confirm

There are some deviations from production that need clarifying (not sure which one is correct actually):

- **(Now) `Free domain for one year` and `Live chat support` not "ticked" in comparison grid for monthly plans:**

<img width="600" alt="Screenshot 2023-07-03 at 1 09 51 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/8a67e709-a047-4682-ad84-81bcb9ff68e8">

- **(Now) Plan type selector shown above comparison grid (wasn't visible previously):**

<img width="600" alt="Screenshot 2023-07-03 at 1 08 55 PM" src="https://github.com/Automattic/wp-calypso/assets/1705499/6b4e9e3f-7ac0-47cf-ae3d-dd1b4996e5e7">


### TODO

- [x] Tune the default "intents" used or mark with a TODO to inform upcoming work

## Testing Instructions



* Go through `/setup/wooexpress` and grab the site created in the end
* Go to `/plans/[woo site]` and observe/test plans grids work as before (given any deviations mentioned above)

## Pre-merge Checklist
